core/src/main/scala/caliban/validation/FragmentValidator.scala

```diff
@@ -113,9 +113,9 @@ object FragmentValidator {
       }
 
     val conflicts = sameResponseShapeByName(selectionSet) ++ sameForCommonParentsByName(selectionSet)
-
-    IO.whenCase(conflicts) { case Chunk(head, _*) =>
-      IO.fail(ValidationError(head, ""))
-    }.unit
+    conflicts match {
```

Comment:
Avoid a "a type was inferred to be `Any`; this may indicate a programming error." when using `ZIO.whenCase`